frontend/src/utils/Common.ts

```diff
@@ -17,6 +17,22 @@ export const arrayEquals = <T>(a1: T[], a2: T[], comparator: (v1: T, v2: T) => b
 export const namespaceEquals = (ns1: Namespace[], ns2: Namespace[]): boolean =>
   arrayEquals(ns1, ns2, (n1, n2) => n1.name === n2.name);
 
+export const namespacesPerCluster = (
```

Comment:
```suggestion
// namespacesPerCluster takes an array of <namespaceNames> and an array of <namespaces> (with cluster info) 
// and returns the subset of namespaceNames that are found on the target cluster. If <namespaces> is
// undefined all <namespaceNames> are returned.
export const namespacesPerCluster = (
```

Minimally, I think this utility could use a comment, because it's not really clear what it's doing or what it expects.  I think also, `namespacesForCluster` would be better that `namespacesPerCluster`.

But maybe more than that, it may be over-complicated because I think you could use `store.getState().namespaces.namespacesPerCluster` to immediately get the namespaces for the target cluster, and then filter them by the activeNamespace names.

Reply:
Have renamed the method.
However the `store.getState().namespaces.namespacesPerCluster` is empty, as this variable is set outside of the graph. But let me check if I can fix that and reuse.

Reply:
hmm, I thought that was set at login.  Note that I think JSON.stringify of a map just shows `{}`, even if the map has data.  Anyway, if I'm wrong then what you have should be fine.